ProcessOptimizer/samplers/doe/doe_utils.py

```diff
@@ -0,0 +1,146 @@
+import warnings
+
+import numpy as np
+
+
+def generate_replicas_and_sort(
+    design_points_real_space, n_replicates, sorting=False
```

Comment:
In general, I would prefer type hints on all inputs and outputs.
```suggestion
    design_points_real_space: Union[list, np.ndarray], n_replicates: int, sorting: Union[bool, str]=False
```

Reply:
On the todo list